lib/Driver/Driver.cpp

```diff
@@ -2031,6 +2031,12 @@ const ToolChain *Driver::getToolChain(const ArgList &Args) const {
       TC = new toolchains::Darwin(*this, Target);
       break;
     case llvm::Triple::Linux:
+      if (Target.isAndroid()) {
```

Comment:
I think that we should specialize GenericUnix into a GNUToolchain or LinuxToolchain and sink the android check into the toolchain.  Android is an environment of linux, and the toolchain really is identical across the two.  Creating two toolchains feels wrong.  This has previously been a problem for modeling windows in clang, and it was a fair amount of effort to clean up afterwards.  I think we should draw on our experiences there to improve our choices here.